File: 문제풀이/chap2-5/prob2-5-4.cpp
#include <iostream>
using namespace std;

bool isprime(int x) {
	for (int i = 2; i <= x - 1; i++) {
		// x를 i로 나눈 나머지가 0이라면, x는 i로 나누어짐
		if (x % i == 0) return false;
	}
	return true;
}

int main() {
	// 입력
	int N;
	cin >> N;
	
	// 답 출력(공백 문자로 구분해서 출력하는 경우)
	for (int i = 2; i <= N; i++) {
		if (isprime(i) == true) {
			if (i >= 3) cout << " ";
			cout << i;
		}
	}
	cout << endl;
	return 0;
}

File: 문제풀이/chap2-5/prob2-5-4.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		
		// 입력
		int N = sc.nextInt();
		
		// 출력
		for (int i = 2; i <= N; i++) {
			if (isPrime(i) == true) {
				if (i >= 3) System.out.print(" ");
				System.out.print(i);
			}
		}
		System.out.println();
	}
	static boolean isPrime(long N) {
		// 2 이상의 정수 N을 매개변수로 받고, N이 소수라면 True를, 아니라면 False를 리턴하는 함수
		for (long i = 2; i <= N - 1; i++) {
			if (N % i == 0) {
				return false;
			}
		}
		return true;
	}
}